src/NServiceBus.Transport.Msmq/MsmqMessageDispatcher.cs

```diff
@@ -77,6 +79,8 @@ void ExecuteTransportOperation(TransportTransaction transaction, UnicastTranspor
 
                         var label = GetLabel(message);
 
+                        await Task.Yield();
```

Comment:
You could move this call to the top of the method instead, and then the entire method body would be executed in the continuation.